packages/core/src/data_sources/index.ts

```diff
@@ -148,4 +149,59 @@ export default class DataSourceManager extends ItemManagerModule<ModuleConfig, D
 
     return result;
   }
+
+  /**
+   * Store data sources to a JSON object.
+   * @returns {Object} Stored data sources.
+   */
+  store() {
+    const data: ObjectAny = {};
+    this.all.forEach((dataSource) => {
+      const shouldStoreInProject = dataSource.get('shouldStoreInProject');
+      if (shouldStoreInProject) {
+        data[dataSource.id] = {
+          id: dataSource.id,
+          name: dataSource.get('name' as any),
+          records: dataSource.records.toJSON(),
+          shouldStoreInProject,
+        };
+      }
+    });
+
+    return { [this.storageKey]: data };
```

Comment:
It's a bit weird that we're storing data sources in an object and other storable modules are in an array. Any reason behind it?
Storing arbitrary strings in objects is also a common issue, eg. if the `id` is `constructor` it won't be saved properly 🥲

Reply:
I made it a key value simply so we can retrieve the data source by id faster than iterating over an array, it should not be too much of a performance problem so I am happy to change it to array like others.